### DISCUSSION TITLE
Push and pull from github repo, from another website

### DISCUSSION THREAD

**Discussion by Panyeczky** [community/community#25304] [Actions]
<p>Hey Everyone!</p>
<p>I’ve got a task at my workspace which is I have to manage the insider tools of the company.<br>
I have to label them, create a searching method for them, so the main ideal is to create a website. Pulling should be a main feature as well.</p>
<p>So the main issue, and my question is: is there <strong>any</strong> way, to pull branches from another website then the official github repo page?</p>

**Comment by airtower-luna** [comment]
<p>What do you mean by “pulling branches” in the context of a website? In general you can automate most things using the API.</p>

**Comment by Panyeczky** [comment]
<p>I would like to get the same result as if I would type “git pull” in the console after choosing the branch, by pressing a button on a website.</p>

**Comment by airtower-luna** [comment]
<p>If you mean modifying your local copy of a Git repository, that’s not possibly to do from a website, because the website can’t just write to your filesystem. If you want to manage your repository using a graphical interface you need install a Git client that has one. I don’t use any myself, but there are a lot of options, see:</p>
<ul>
<li><a href="https://git-scm.com/book/en/v2/Appendix-A%3A-Git-in-Other-Environments-Graphical-Interfaces" class="inline-onebox">Git - Graphical Interfaces</a></li>
<li><a href="https://git-scm.com/downloads/guis" class="inline-onebox">Git - GUI Clients</a></li>
</ul>

**Comment by Panyeczky** [comment]
<p>Is there an option to download a copy of Git repository as a zip, by using the API or anything else?</p>

**Comment by airtower-luna** [comment]
<p>See here: <a href="https://docs.github.com/en/rest/reference/repos#download-a-repository-archive-zip" class="inline-onebox">Repositories - GitHub Docs</a></p>
<p>Though you might be better of actually using Git, instead of downloading everything each time.</p>

**Comment by Panyeczky** [comment, marked as answer]
<p>Final result was opening terminal via php and executing a script which pull the repository.<br>
Thanks for helping!</p>
<p>Best regards!</p>
